test/feature-benchmark/scenarios.py

```diff
@@ -803,3 +803,63 @@ class KafkaRecovery(KafkaScenario):
 true
 """
     )
+
+
+class KafkaRecoveryBig(Kafka):
+    SHARED = [
+        Td("$ kafka-create-topic topic=kafka-recovery-big partitions=8"),
+        # Ingest 10M records
+        Kgen(
+            topic="kafka-recovery-big",
+            args=[
+                "--keys=random",
+                "--num-records=10000000",
+                "--values=bytes",
+                "--max-message-size=32",
+                "--min-message-size=32",
+                "--key-min=256",
+                "--key-max=10000000",
+            ],
+        )
+        # Add 256 EOF markers
+        ,
+        Kgen(
+            topic="kafka-recovery-big",
+            args=[
+                "--keys=sequential",
+                "--num-records=256",
```

Comment:
Why 256? (Just a hint for a bit more commentary in the source here, please. It'd also be worthwhile to explicitly point out that the two kgen calls intentionally use disjoint sets of keys.)

Reply:
Yes, will do. I picked 256 so that even if you have a lot of partitions, each partition will be assigned at least one of those keys and therefore the benchmark will wait until all the partitions have been fully ingested.